src/lib/SocialIcon.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+  import Fa from 'svelte-fa/src/fa.svelte'
+  import { iconMapping } from './iconMapping'
+
+  export let social: keyof typeof iconMapping = 'Website'
+  export let id: string | undefined = undefined
+  export let link: string | undefined = undefined
+  let titleOverwrite: string | undefined = undefined
+  export let noLink = false
+  let finalLink = link
+
+  $: {
+    if (!noLink && !link && id) {
+      switch (social) {
+        case 'Discord':
+          finalLink = `https://discord.gg/${id}`
+          break
+        case 'Twitter/X':
+          finalLink = `https://twitter.com/${id}`
+          break
+        case 'Facebook':
+          finalLink = `https://facebook.com/${id}`
+          break
+        case 'Instagram':
+          finalLink = `https://instagram.com/${id}`
+          break
+        case 'Twitch':
+          finalLink = `https://twitch.tv/${id}`
+          break
+        case 'BeatSaver':
+          finalLink = `https://beatsaver.com/profile/${id}`
+          break
+        case 'YouTube':
```

Comment:
Are these supposed to be empty?

Reply:
Yeah! In these cases you should just specify the link. Maybe it could be done differently for YouTube, Reddit and GitHub, but I wasn't sure if special cases arise.